sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -339,21 +340,20 @@ export class EventHubSender extends LinkEntity {
     const timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
     retryOptions.timeoutInMs = timeoutInMs;
 
-    const initStartTime = Date.now();
-    await this._createLinkIfNotOpen(options);
-    const timeTakenByInit = Date.now() - initStartTime;
-
     const sendEventPromise = async (): Promise<void> => {
+      const initStartTime = Date.now();
+      const sender = await this._createLinkIfNotOpen(options);
```

Comment:
I could update the code everywhere we call `_createLinkIfNotOpen` to also set the private variable. I think that's only 2 places currently so it's not too much burden to change.

Reply:
Oooh, in that case how about renaming `_createLinkIfNotOpen()` to `getLink()` in that case? Because in this PR we are indeed changing the semantics of this function. Previously it was "if open, do nothing, else create link" and now it is "get me a/the link"

Reply:
We now have `_getLink()`

Reply:
After seeing the changes where we are now setting this._sender in multiple places, it does look more weird than what we had before. Sigh, I take my objection back.

Reply:
I should have held my ground :smile: But sometimes you don't realize how weird it looks until it exists. Updated.